cmd/infra/aws/create_cli_role.go

```diff
@@ -0,0 +1,246 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/spf13/cobra"
+
+	awsutil "github.com/openshift/hypershift/cmd/infra/aws/util"
+	"github.com/openshift/hypershift/cmd/log"
+)
+
+const (
+	cliRolePolicy = `{
```

Comment:
Didn't @muraee just add this?

Reply:
yes he did, This PR wasn't meant to be reviewed yet (but I always welcome an early review :) ) and I was using his work for testing. So this comment and the following comments should have been left on his pr. In this case Im more than happy to make any changes that @muraee approves.

Reply:
OK. His PR already went in, let's get these following comments addressed in this PR then.